### DISCUSSION TITLE
Would we be able to queue tasks in the future?

### DISCUSSION THREAD

**Discussion by pandamega** [nianeyna/ao3downloader#165] [Ideas]
I was wondering if it would be possible to queue several tasks for the downloader to do. I like to run it overnight (since it can sometimes take a long time and I can't really use AO3 while its running doe to all the breaks), but it would be great if I could tell it to do several things overnight. For example, I could tell it to download fics from multiple links (eg. bookmarks, authors I follow, and tags I'm following) as well as get a list of the fic links, and a spreadsheet with the metadata for each fic, and then also update works in a series, and then check everything for updates. That way everything would be complete in the morning, instead of having to enter a new command every time it finishes a task. I really love this program, and if it's possible to add this feature it would be amazing!

**Comment by nianeyna** [comment]
It should be theoretically possible to queue multiple tasks by writing a batch or shell script, but I understand that most people don't want to deal with that haha. Especially since ao3downloader isn't really well-designed to be invoked programmatically. This is a pretty reasonable ask though, so I will try to think of a good way to implement it - I am going to create an issue from this discussion, which you can subscribe to in order to get notified when it is implemented (or of any other developments).

In the meantime, if you *just* want to download multiple links in a row, and you don't need to include updating or any other functions, you should be able to do so using the 'download links from file' option (I know the readme says that only works for work and series links, but it actually will do listing pages as well, just as if you'd entered them manually)
